packages/select-rich/src/LionSelectRich.js

```diff
@@ -369,6 +378,31 @@ export class LionSelectRich extends SlotMixin(ScopedElementsMixin(OverlayMixin(L
     this._overlayCtrl.removeEventListener('hide', this.__overlayOnHide);
   }
 
+  /**
+   * Align invoker width with content width
+   * Make sure display is not set to "none" while calculating the content width
+   */
+  _alignInvokerWidth() {
+    if (this._overlayCtrl && this._overlayCtrl.content) {
+      setTimeout(() => {
+        const initContentDisplay = this._overlayCtrl.content.style.display;
+        const initContentMinWidth = this._overlayCtrl.content.style.minWidth;
+        const initContentWidth = this._overlayCtrl.content.style.width;
+
+        this._overlayCtrl.content.style.display = '';
+        this._overlayCtrl.content.style.minWidth = 'auto';
+        this._overlayCtrl.content.style.width = 'auto';
+
+        const contentWidth = this._overlayCtrl.content.getBoundingClientRect().width;
+        this._invokerNode.style.width = `${contentWidth + this.arrowWidth}px`;
+
+        this._overlayCtrl.content.style.display = initContentDisplay;
+        this._overlayCtrl.content.style.minWidth = initContentMinWidth;
+        this._overlayCtrl.content.style.width = initContentWidth;
+      });
+    }
+  }
```

Comment:
```suggestion
  async _alignInvokerWidth() {
    if (this._overlayCtrl && this._overlayCtrl.content) {
      await this.updateComplete;
      const initContentDisplay = this._overlayCtrl.content.style.display;
      const initContentMinWidth = this._overlayCtrl.content.style.minWidth;
      const initContentWidth = this._overlayCtrl.content.style.width;

      this._overlayCtrl.content.style.display = '';
      this._overlayCtrl.content.style.minWidth = 'auto';
      this._overlayCtrl.content.style.width = 'auto';

      const contentWidth = this._overlayCtrl.content.getBoundingClientRect().width;
      this._invokerNode.style.width = `${contentWidth + this.arrowWidth}px`;

      this._overlayCtrl.content.style.display = initContentDisplay;
      this._overlayCtrl.content.style.minWidth = initContentMinWidth;
      this._overlayCtrl.content.style.width = initContentWidth;
    }
  }
```

That works for me in the demo, and it's much faster than doing a setTimeout, because promises are queued as microtasks so they happen faster. And it also means that if updateComplete is already resolved, no time is lost.